### PR TITLE
[Quest]A Bitter Past

### PR DESCRIPTION
**Pull request by hooksta4**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
- Implements quest A Bitter Past (bring a tissue its a sad one!)
- Adds Yinyang Lorgnette to Item.lua
- Adds defaultActions based on [capture](https://www.youtube.com/watch?v=koHiHNHhqRk)
- Adjusts NM's to be flagged as NM
- Adjusts NM spell list (Beastmen DRK / BLM respectively)
- Adds NM lua scripts (blank)
- Verifies Frescheque default dialog and adjusts
- Adds dialog and mob to lufaise ID's.lua

Allows players to complete the quest A Bitter Past.
## Steps to test these changes
Speak with Frescheque to begin the quest.

### CODE REVIEW

File: scripts/zones/Lufaise_Meadows/IDs.lua [hunk 1]
```diff
@@ -41,6 +42,7 @@ zones[xi.zone.LUFAISE_MEADOWS] =
         SPLINTERSPINE_GRUKJUK = GetFirstID('Splinterspine_Grukjuk'),
         KURREA                = GetFirstID('Kurrea'),
         AMALTHEIA             = GetFirstID('Amaltheia'),
+        BITTER_PAST_MOBS      = GetFirstID('Blackbone_Frazdiz'),
```
Comment (zach2good) [inline scripts/zones/Lufaise_Meadows/IDs.lua]: BLACKBONE_FRAZDIZ

Comment (zach2good) [reply scripts/zones/Lufaise_Meadows/IDs.lua]: To make the different mobs readable, just extract the IDs locally at the top of the quest file

File: scripts/quests/otherAreas/A_Bitter_Past.lua [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        [xi.zone.LUFAISE_MEADOWS] =
+        {
+            ['qm_bitter_past'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        quest:getVar(player, 'nmKilled') ~= 1 and
+                        npcUtil.popFromQM(player, npc, {lufaiseID.mob.BITTER_PAST_MOBS, lufaiseID.mob.BITTER_PAST_MOBS + 1},
+                        {claim = true, hide = 0})
```
Comment (zach2good) [inline scripts/quests/otherAreas/A_Bitter_Past.lua]: Find a better way to format this please

File: scripts/quests/otherAreas/A_Bitter_Past.lua [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                end,
+            },
+
+            ['Rainbringer_Yjatvot'] =
+            {
+                onMobDeath = function(mob, player)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        GetMobByID(lufaiseID.mob.BITTER_PAST_MOBS):isDead()
+                    then
```
Comment (zach2good) [inline scripts/quests/otherAreas/A_Bitter_Past.lua]: BITTER_PAST_MOBS no longer exists

Comment (hooksta4) [reply scripts/quests/otherAreas/A_Bitter_Past.lua]: Good point.

**Review by zach2good — APPROVED** [review]

File: scripts/quests/otherAreas/A_Bitter_Past.lua [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+-- Log ID: 4, Quest ID: 66
+-- Frescheque : !pos 18 -36 12 26
+-- Raminey    : !pos 82 -35 50 26
+-- Equette    : !pos 3 -22 -17 26
+-- ???        : !pos 58 -7 27 24
+-----------------------------------
+local lufaiseID = zones[xi.zone.LUFAISE_MEADOWS]
+-----------------------------------
+local quest  = Quest:new(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.A_BITTER_PAST)
+local orcNM  = lufaiseID.mob.BLACKBONE_FRAZDIZ
```
Comment (Xaver-DaRed) [inline scripts/quests/otherAreas/A_Bitter_Past.lua]: Theres unnecesary double space here between var name and =

File: scripts/quests/otherAreas/A_Bitter_Past.lua [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStaus.QUEST_COMPLETED and
+                quest:getVar(player, 'Prog') == 3
+        end,
```
Comment (Xaver-DaRed) [inline scripts/quests/otherAreas/A_Bitter_Past.lua:140]: quest:complete deletes quest vars iirc. This will never be 3

Comment (hooksta4) [reply scripts/quests/otherAreas/A_Bitter_Past.lua:140]: Yea - let me double check on this

**Review by Xaver-DaRed — APPROVED** [review]

**Pull request merged**